azurelinuxagent/common/utils/restutil.py

```diff
@@ -112,15 +114,19 @@ def http_request(method, url, data, headers=None, max_retry=3,
 
     # If httplib module is not built with ssl support. Fallback to http
     if secure and not hasattr(httpclient, "HTTPSConnection"):
-        logger.warn("httplib is not built with ssl support")
         secure = False
+        if secure_warning:
+            logger.warn("httplib is not built with ssl support")
+            secure_warning = False
```

Comment:
@hglkrijger I can see you set it `False` here, but where does it ever achieve `True`?